Python/codecs.c

```diff
@@ -16,6 +16,27 @@ Copyright (c) Corporation for National Research Initiatives.
 #include "pycore_pystate.h"       // _PyInterpreterState_GET()
 #include "pycore_ucnhash.h"       // _PyUnicode_Name_CAPI
 
+#define CODECS_STRICT_ERROR_POLICY              0
+#define CODECS_IGNORE_ERROR_POLICY              1
+#define CODECS_REPLACE_ERROR_POLICY             2
+#define CODECS_XMLCHARREFREPLACE_ERROR_POLICY   3
+#define CODECS_BACKSLASHREPLACE_ERROR_POLICY    4
+#define CODECS_NAMEREPLACE_ERROR_POLICY         5
+#define CODECS_SURROGATEPASS_ERROR_POLICY       6
+#define CODECS_SURROGATEESCAPE_ERROR_POLICY     7
+#define CODECS_ERROR_POLICY_COUNT               8
+
+static const char codecs_native_error_polcies[CODECS_ERROR_POLICY_COUNT][32] = {
```

Comment:
I suggest a simpler array:

```suggestion
static const char* codecs_native_error_policies[] = {
    "strict",
    "ignore",
    ...
```

Reply:
I think I got a compiler error because of that because of some initialization being not constant anymore (I don't want to duplicate those policies in the initialization but the following code failed to compile:

```C
static struct {
	const char *name;
    PyMethodDef def;
} methods[] = {
	{
		codecs_native_error_policies[0], 
		{ ... },
	}
};
```

I'll check it again tomorrow (maybe I miswrote something).

Reply:
Yes, I cannot use the above construct since I get:

```text
Python/codecs.c: In function ‘_PyCodec_InitRegistry’:
Python/codecs.c:1428:13: error: initializer element is not constant
             codecs_native_error_polcies[0],
```

Reply:
Resolved by reverting to the old code.